File: apps/factory-web/src/features/tool-change/domain/toolChange.ts
export interface ToolChangeProvenance {
  source: { kind: "REAL"; provider: "NIST"; sourceSetId: string; artifactId: string };
  transformation: { rawRecordId: string; mappingVersion: string; sourceDataItemId: string };
}

export interface ToolChange {
  replaySequence: number;
  sourceObservedAt: string;
  fromToolNumber: number;
  toToolNumber: number;
  provenance: ToolChangeProvenance;
}

export interface ToolChangeTimeline {
  schemaVersion: "1.0.0";
  machineId: string;
  replaySessionId: string;
  throughReplaySequence: number;
  toolChanges: ToolChange[];
}

File: apps/factory-api/src/main/java/com/forgesync/factoryapi/toolchange/domain/ToolChange.java
package com.forgesync.factoryapi.toolchange.domain;

import java.time.Instant;
import java.util.UUID;

public record ToolChange(
    UUID replaySessionId,
    long replaySequence,
    Instant sourceObservedAt,
    long fromToolNumber,
    long toToolNumber,
    String sourceDataItemId,
    String sourceSetId,
    String artifactId,
    String rawRecordId,
    String mappingVersion) {}
